### PR TITLE
Backend: add argument to `fileDiffs` to filter diffs to a file or directory

### PR DESCRIPTION
**Pull request by camdencheek**
This adds an argument to `fileDiffs` so that we can filter the resulting diffs to a set of files or directories. This should be the extent of the backend changes required to implement a view of the changes for a single file as described [here](https://github.com/sourcegraph/sourcegraph/issues/13411)

Examples: 

<details>
<summary>Get change in `dev/gqltest/search_test.go` over the last 50 commits</summary>

```graphql
{
  repository(name: "github.com/sourcegraph/sourcegraph") {
    name
    comparison(base:"HEAD~50", head:"HEAD") {
      fileDiffs(paths:["dev/gqltest/search_test.go"]){
        nodes{
          hunks {
            body
          }
        }
      }
    }
  }
}
```

</details>

<details>
<summary>Get all commits that touch the directory `internal/search/job/jobutil` and the diff associated with them</summary>

```graphql
{
	repository(name:"github.com/sourcegraph/sourcegraph") {
    commit(rev:"HEAD"){
      ancestors(path:"internal/search/job/jobutil") {
        nodes {
          diff{
            fileDiffs(paths:["internal/search/job/jobutil"]){
              nodes{
                hunks {
                  body
                }
              }
            }
          }
        }
      }
    }
  }
}
```

</details>

## Test plan

Added unit test, tested manually to make sure git stuff works as expected.

### CODE REVIEW

File: doc/code_monitoring/explanations/core_concepts.md [hunk 1]
```diff
@@ -8,17 +8,18 @@ Code monitors are made up of two main elements: **Triggers** and **Actions**.
 
 A _trigger_ is an event which causes execution of an action. Currently, code monitoring supports one kind of trigger: "When new search results are detected" for a particular search query. When creating a code monitor, users will be asked to specify a query as part of the trigger.
 
-Sourcegraph will run the search query periodically, and when new results for the query are detected, a trigger event is executed. In response to the trigger event, any _actions_ attached to the code monitor will be executed.
```
Comment (eseliger) [inline doc/code_monitoring/explanations/core_concepts.md]: is this related?

**Review by eseliger — APPROVED** [review]
awesome :)

File: doc/code_monitoring/explanations/core_concepts.md (hunk 1 above)
Comment (camdencheek) [reply doc/code_monitoring/explanations/core_concepts.md]: Nope, caught a commit from another branch 😄 Should be fixed now

**Comment by camdencheek** [comment]
FYI, I pushed one more commit that adds the ability to do a diff from a `GitCommitResolver` (returning the same `RepositoryComparison` type). It just makes a bunch of queries way more convenient so callers don't have to make followup requests.

**Review by eseliger — APPROVED** [review]
Still LGTM 😬
